inc/features/blocks/namespace.php

```diff
@@ -69,3 +78,248 @@ function get_block_settings( string $name ) : ?array {
+					'required' => true,
+					'type' => 'string',
+					'validate_callback' => __NAMESPACE__ . '\\validate_id',
+					'sanitize_callback' => __NAMESPACE__ . '\\sanitize_id',
+				],
+				'post_id' => [
+					'description' => __( 'An optional post ID to filter by.', 'altis-experiments' ),
+					'type' => 'number',
+				],
+			],
+		],
+		'schema' => [
+			'type' => 'object',
+			'properties' => [
+				'total' => [ 'type' => 'number' ],
+				'audiences' => [
+					'type' => 'array',
+					'items' => [
+						'type' => 'object',
+						'properties' => [
+							'id' => [ 'type' => 'number' ],
+							'count' => [ 'type' => 'number' ],
+						],
+					],
+				],
+				'posts' => [
+					'type' => 'array',
+					'items' => [
+						'type' => 'object',
+						'properties' => [
+							'id' => [ 'type' => 'number' ],
+							'count' => [ 'type' => 'number' ],
+							'audiences' => [
+								'type' => 'array',
+								'items' => [
+									'type' => 'object',
+									'properties' => [
+										'id' => [ 'type' => 'number' ],
+										'count' => [ 'type' => 'number' ],
+									],
+								],
+							],
+						],
+					],
+				],
+				'postId' => [ 'type' => 'number' ],
+			],
+		],
+	] );
+}
+
+/**
+ * Validate Experience Block ID.
+ *
+ * @param string $param The Experience Block ID.
+ * @return bool
+ */
+function validate_id( $param ) : bool {
+	return (bool) preg_match( '/[a-z0-9-]+/', $param );
+}
+
+/**
+ * Sanitize Experience Block ID.
+ *
+ * @param string $param The Experience Block ID.
+ * @return string
+ */
+function sanitize_id( $param ) : string {
+	return preg_replace( '/[^a-z0-9-]+/', '', $param );
+}
+
+/**
+ * Check current user can view XB analytics data.
+ *
+ * @return boolean
+ */
+function check_views_permission() : bool {
+	return true;
+	$type = get_post_type_object( Audiences\POST_TYPE );
+	return current_user_can( $type->cap->edit_posts );
+}
+
+/**
+ * Retrieve the Experience Block views data.
+ *
+ * @param WP_REST_Request $request The REST request object.
+ * @return WP_REST_Response
+ */
+function handle_views_request( WP_REST_Request $request ) : WP_REST_Response {
+	$block_id = $request->get_param( 'id' );
+	$post_id = $request->get_param( 'post_id' ) ?? null;
+	$views = get_views( $block_id, $post_id );
+	return rest_ensure_response( $views );
+}
+
+/**
+ * Get the Experience Block views data.
+ *
+ * @param string $block_id The Experience block client ID to get data for.
+ * @param int|null $post_id An optional post ID to limit results by.
+ * @return array|WP_Error
+ */
+function get_views( string $block_id, ?int $post_id = null ) {
+	$query = [
+		'query' => [
+			'bool' => [
+				'filter' => [
+					// Set current site.
+					[
+						'term' => [
+							'attributes.blogId.keyword' => get_current_blog_id(),
+						],
+					],
+
+					// Set block ID.
+					[
+						'term' => [
+							'attributes.clientId.keyword' => $block_id,
+						],
+					],
+
+					// Last month.
```

Comment:
Specifically from now minus a month. Should this be pegged to month start/end instead? What about 28/31 day months?

Reply:
It's not all that important really, just a way to capture a set time frame. I think it probably wants to be set to the same amount of time as the data retention value. I should make sure that's noted in the UI really.

Reply:
I've removed the time range and added text to say it's based on the data retention settings. I'll add an issue for the analytics plugin to provide a function for extracting that value so we can show it in the UI.

It's not ideal right now but will be improved upon in a follow up.